base/uk.ac.stfc.isis.ibex.scriptgenerator/src/uk/ac/stfc/isis/ibex/scriptgenerator/pythoninterface/PythonInterface.java

```diff
@@ -133,6 +133,27 @@ public PythonInterface(String actionLoaderPythonScript) {
 		}
 	};
 	
+	/**
+	 * When python has become not ready handle this by trying to restart it.
+	 * MUST always be called in Py4J worker thread.
+	 * MUST never wait for a thread to complete or return in here otherwise deadlock will occur.
+	 */
+	private void handlePythonReadinessChangeNotReadyHelper() {
```

Comment:
Can the method name be changed to something simpler and generic like `restartPython`? I think it would be easier to read if this function is called elsewhere if needed.

Reply:
Yeah the name is not the best actually, changes now to restartPython.